### PR TITLE
Use domain_id for identification if provided in DistributedClosestPoint.

### PR DESCRIPTION
**Pull request by gunney1**
If not provided, use local domain's iteration index. Fix error setting domain.

# Summary

- This PR is a feature
- It does the following (modify list as needed):
  - Modifies `DistributedClosestPoint` to use the blueprint `state/domain_id` field to identify mesh domain, if the field is given.
  - Fixes https://github.com/LLNL/axom/issues/1197
  - Documents this change in the release notes.

Issue https://github.com/LLNL/axom/issues/1197 also requires the same change in `MarchingCubes`, and that change occurred in a separate, merged PR.

This PR and related comments led to issue https://github.com/LLNL/axom/issues/1236, on being bit width compatibility of blueprint data.

### CODE REVIEW

**Comment by rhornung67** [comment]
@gunney1 when this is ready to merge, please confirm in a comment or above in the PR summary that you have manually tested this on blueos and EAS3 systems and verified that all checks have passed. Thank you.

**Comment by gunney1** [comment]
This branch passes the test on blueos (lassen) and EAS-3 (rzvernal and tioga) systems.  (Release build fails on rzansel, but that's happening on the develop branch as well.)

**Comment by gunney1** [comment]
The rzansel tests failures for release build have been resolved this morning.

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 1]
```diff
@@ -455,9 +455,16 @@ class DistributedClosestPointImpl
     axom::Array<axom::IndexType> domIds(ptCount, ptCount);
     std::size_t copiedCount = 0;
     conduit::Node tmpValues;
-    for(conduit::index_t d = 0; d < mdMeshNode.number_of_children(); ++d)
+    for(axom::IndexType d = 0; d < mdMeshNode.number_of_children(); ++d)
```
Comment (kennyweiss) [inline src/axom/quest/DistributedClosestPoint.hpp:458]: I looked into this recently and `number_of_children()` returns a `conduit::index_t`, which is a signed int (either 32-bits or 64-bits based on how conduit is configured)

`axom::IndexType` is also a signed int32 or int64, based on how axom is configured.
However, there's no guarantee that `conduit::index_t` has the same bitwidth as `axom::IndexType`.

In practice, we're unlikely to run problems with billions of ranks, but if we're explicitly casting to a 32-bit int a few lines down, we should probably be consistent throughout this loop.

See: 
* https://github.com/LLNL/conduit/blob/f570d0dc4a77861ea129b2f5965768f77ce68030/src/libs/conduit/conduit_bitwidth_style_types.h.in#L330-L336
* https://github.com/LLNL/conduit/blob/f570d0dc4a77861ea129b2f5965768f77ce68030/src/libs/conduit/conduit_core.hpp#L63-L68
* https://github.com/LLNL/axom/blob/f511a2593369043c2571f0646fb0ce421acbc572/src/axom/core/Types.hpp#L63-L67

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 2]
```diff
@@ -455,9 +455,16 @@ class DistributedClosestPointImpl
...
     {
       const conduit::Node& domain = mdMeshNode.child(d);
+
+      axom::IndexType domainId = d;
+      if(domain.has_path("state/domain_id"))
+      {
+        domainId = domain.fetch_existing("state/domain_id").to_int32();
+      }
```
Comment (kennyweiss) [inline src/axom/quest/DistributedClosestPoint.hpp:466]: Minor: Should we add a check that the `domain_id`s are distinct? Perhaps in a `verify` function?

I can imagine the algorithm behaving badly if the domains are not distinct.

File: src/axom/quest/examples/quest_distributed_distance_query_example.cpp [hunk 3]
```diff
@@ -412,10 +412,13 @@ struct BlueprintParticleMesh
...
-  void setPoints(const axom::Array<primal::Point<double, NDIMS>>& pts)
+  void setPoints(axom::IndexType domainId,
+                 const axom::Array<primal::Point<double, NDIMS>>& pts)
   {
-    axom::IndexType domainIdx = createBlueprintStubs();
-    SLIC_ASSERT(m_domainGroups[domainIdx] != nullptr);
+    axom::IndexType localIdx = createBlueprintStubs();
+    SLIC_ASSERT(m_domainGroups[localIdx] != nullptr);
+    m_domainGroups[localIdx]->createViewScalar<std::int64_t>("state/domain_id",
+                                                             domainId);
```
Comment (kennyweiss) [inline src/axom/quest/examples/quest_distributed_distance_query_example.cpp:421]: Minor: This function is also (potentially) mixing 32-bit and 64-bit ints w/ the conversion from `axom::IndexType` to `std::int64_t`

@cyrush -- when the blueprint refers to an `integer`, is there an assumption about its bitwidth?
See: https://llnl-conduit.readthedocs.io/en/latest/blueprint_mesh.html#state

**Review by kennyweiss — APPROVED** [review]
Thanks @gunney1 -- I just had a few comments about (possible) implicit conversions between 32- and 64-bit ints.

**Review by rhornung67 — APPROVED** [review]

**Review by agcapps — APPROVED** [review]
Thanks, Brian.

**Pull request merged**